src/ZkFold/Base/Protocol/ARK/Plonk.hs

```diff
@@ -117,13 +117,52 @@ instance Arbitrary PlonkProverSecret where
         arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary
         <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary
 
-newtype PlonkInput = PlonkInput (V.Vector F)
+newtype PlonkInput = PlonkInput F
     deriving (Show)
 
+instance Arbitrary PlonkInput where
+    arbitrary = do
+        PlonkInput . negate <$> arbitrary
+
 data PlonkProof = PlonkProof G1 G1 G1 G1 G1 G1 G1 G1 G1 F F F F F F
     deriving (Show)
 
--- TODO (Issue #18): make the code safer, check list lengths (?)
+commonSetup :: forall d .
```

Comment:
We do not need this function since the setup should be run only once per circuit. But the `prove` algorithm can be run many times with different inputs.